rust/src/treefile.rs

```diff
@@ -2755,7 +2755,12 @@ impl TreeComposeConfig {
     fn hasher_update(&self, hasher: &mut glib::Checksum) -> Result<()> {
         // don't use pretty mode to increase the chances of a stable serialization
         // https://github.com/projectatomic/rpm-ostree/pull/1865
-        hasher.update(serde_json::to_vec(self)?.as_slice());
+        let layers = serde_json::to_string(&self.base.ostree_layers.clone()).unwrap();
+        let mut j = serde_json::to_string(self)?;
```

Comment:
Another option here is to do e.g.:

```
    let self_copy = self.clone();
    self_copy.ostree_layers = None;
    hasher.update(serde_json::to_vec(&self_copy)?.as_slice());
```

or so?

Reply:
self_copy is still a '&' reference to an immutable object and hence assigning None to its field will result in an error

Reply:
OK right, we need to do `(*self).clone()`.

However, doing that requires https://github.com/coreos/rpm-ostree/pull/3888